providers/mlx5/cq.c

```diff
@@ -1070,6 +1070,8 @@ static inline void _mlx5_end_poll(struct ibv_cq_ex *ibcq,
 	if (lock)
 		mlx5_spin_unlock(&cq->lock);
 
+	cq->cur_rsc = NULL;
```

Comment:
We already have that code (i.e. cq->cur_rsc = NULL)  as part of mlx5_start_poll(), so it seems redundant.

Note:
In any case, it had to be done before line 1070 to be done under the lock, if the lock was applicable.

Reply:
Got it. I have just removed this line in the latest patch.